services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,62 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  if (event == 0){
+    return ERR_CODE_INVALID_ARG;
+  }
+
   /* Send an event to the thermal manager queue */
+  if (xQueueSend(thermalMgrQueueHandle, event, (TickType_t)5) != pdPASS){
+    return ERR_CODE_UNKNOWN; // failed to send message
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = {};
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event); //OS interrupt occurred, send OS interrupt event
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+
+  thermal_mgr_event_t buffer = {};
+
+  float tempC = 0;
+
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &buffer, (TickType_t) 5) != pdPASS){ 
+      continue; 
+    }  
+    //received next event
+
+    if (buffer.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){ //temperature reading event
+      //read temperature
+      if (readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC) != ERR_CODE_SUCCESS){
+        continue; //some error occurred
+      }
+      
+      addTemperatureTelemetry(tempC); //add to temperature telemetry
+
+      if (tempC >= LM75BD_DEFAULT_OT_THRESH){
+        osHandlerLM75BD(); //must check if current temperature is over T_{th}; if it is, interrupt OS for overtemperature
```

Comment:
Never call an isr function directly. This will be called when the temperature reaches a certain threshold automatically thus this if statement isn't needed either